### PR TITLE
Convert notices to use native nonce generation from WC Admin

### PR DESCRIPTION
**Pull request by roykho**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

So the original issue where people could not dismiss admin notices (i.e. database update notice) was raised here https://github.com/woocommerce/woocommerce/issues/27424

In summary it was due to the fact that stored nonces in the db gets invalidated after a period of time making the notice not dismissible anymore.

Since at the time WC Admin did not have a native way to handle nonces, a work around was proposed and used here https://github.com/woocommerce/woocommerce/pull/28500

WC Admin will eventually support native way to accept and create nonces. Work being done here https://github.com/woocommerce/woocommerce-admin/pull/6726

So this PR is an update to remove the work around solution to utilize the WC Admin's native way of creating nonces.

Thanks @adrianduffell 

### How to test the changes in this Pull Request:

Activate WC Admin as a standalone plugin and checkout this PR https://github.com/woocommerce/woocommerce-admin/pull/6726 along with the one you're reviewing now.

1. Set your WooCommerce database version to something older like (4.9.0) You can find this in the `wp_options` table under `woocommerce_db_version` option name.
2. Reload your WooCommerce dashboard page and you should now see the database update notice.
3. Ensure you're able to click the update button and see that it starts progressing with the new notice.
4. After a small period of time, reload the page and ensure you see the notice that implies the update was done.
5. Click on the `Thanks` button and ensure it dismisses the notice.

### Other information:

This PR should only go in after WC Admin package with this change has been merged.

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Tweak - Use WC Admin's native notice nonce generation.

### CODE REVIEW

**Review by adrianduffell — DISMISSED** [review]
@roykho This looks perfect to me.

Thanks for reviewing the change on the WC Admin side. I'll aim to merge it ahead of our next release pending review from my team. This should see it included in core 5.4.

File: includes/admin/class-wc-admin-notices.php
Comment (claudiosanches) [inline includes/admin/class-wc-admin-notices.php:73]: We can't just remove a public static method.

Comment (claudiosanches) [reply includes/admin/class-wc-admin-notices.php:73]: Use `_deprecated_function( __CLASS__ . '::' . __FUNCTION__, '5.3.0' );`

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Seems to be working just fine, but there's one small issue, we can't remove public methods like that, it's better to make deprecated.

File: includes/admin/class-wc-admin-notices.php
Comment (roykho) [reply includes/admin/class-wc-admin-notices.php:73]: I thought this was internal enough to sneak away.. but yes you're right, I will deprecate this instead.

Comment (roykho) [reply includes/admin/class-wc-admin-notices.php:73]: @claudiosanches - done!

**Review by claudiosanches — APPROVED** [review]
Looks great now!

**Comment by claudiosanches** [comment]
Just noting here that we need to wait before merging this PR, since will require Woo Admin to get updated as well.

**Pull request merged**
